gorilla-minitwit/internal/db/db_methods.go

```diff
@@ -0,0 +1,276 @@
+package db
+
+import (
+	"encoding/hex"
+	"fmt"
+	"gorilla-minitwit/internal/config"
+	"gorilla-minitwit/internal/models"
+	"strconv"
+	"time"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/schema"
+)
+
+func CheckValueInMap(maps []map[interface{}]interface{}, value interface{}) bool {
+	for _, m := range maps {
+		for _, v := range m {
+			if v == value {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func ConnectDB(dsn string) (*gorm.DB, error) {
+	fmt.Println("Connecting to the database...")
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{NamingStrategy: schema.NamingStrategy{SingularTable: true}})
```

Comment:
What about using the URI here instead?

```suggestion
func ConnectDB(uri string) (*gorm.DB, error) {
	db, err := gorm.Open(postgres.Open(uri), &gorm.Config{NamingStrategy: schema.NamingStrategy{SingularTable: true}})
	...
}
```

Reply:
If you look in our DevOps Adventure Repo this is also how we did it there ;)